datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/common/mappers/InstitutionalMemoryMetadataMapper.java

```diff
@@ -18,8 +21,20 @@ public InstitutionalMemoryMetadata apply(@Nonnull final com.linkedin.common.Inst
         final InstitutionalMemoryMetadata result = new InstitutionalMemoryMetadata();
         result.setUrl(input.getUrl().toString());
         result.setDescription(input.getDescription());
-        result.setAuthor(input.getCreateStamp().getActor().toString());
+        result.setAuthor(getAuthor(input.getCreateStamp().getActor().toString()));
         result.setCreated(AuditStampMapper.map(input.getCreateStamp()));
         return result;
     }
+
+    private CorpUser getAuthor(String actor) {
+        CorpUser corpUser = new CorpUser();
+        try {
+            CorpuserUrn corpuserUrn = CorpuserUrn.createFromString(actor);
+            corpUser.setUrn(actor);
+            corpUser.setUsername(corpuserUrn.getUsernameEntity());
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(String.format("Failed to retrieve author with urn %s, invalid urn", actor));
+        }
+        return corpUser;
+    }
```

Comment:
can you follow the pattern from OwnerMapper I linked to construct a corp user entity just with the actor? Then, you can rely on graphql resolvers to fetch properties like username